### PR TITLE
fix: Extend BaseButtonProps to our buttons and export utility functions

### PR DESCRIPTION
**Pull request by mannycarrera4**
## Summary

Fixes: #1746, https://github.com/Workday/canvas-kit/issues/1784

We weren't extending the `BaseButtonProps` which created some typescript errors even though the styles rendered. This change allows for better documentation on how to customize our buttons.

category

### Release Note
- You can now overwrite styles using style properties from `Box` on `PrimaryButton`, `SecondaryButton`, `TertiaryButton` and `DeleteButton`. This allows you to customize styles more easily.
- We're exporting utility functions `getMinWidthStyles` and `getPaddingStyles` to help those making custom buttons

---

### CODE REVIEW

File: modules/react/button/index.ts [hunk 1]
```diff
@@ -9,6 +9,11 @@ export {
 export {ToolbarIconButton, ToolbarIconButtonProps} from './lib/ToolbarIconButton';
 export {ToolbarDropdownButton, ToolbarDropdownButtonProps} from './lib/ToolbarDropdownButton';
 export {Hyperlink, HyperlinkProps} from './lib/Hyperlink';
-export {BaseButton, ButtonContainerProps} from './lib/BaseButton';
+export {
+  BaseButton,
+  ButtonContainerProps,
```
Comment (RayRedGoose) [inline modules/react/button/index.ts:14]: So as you created `BaseButtonProps` can you replace `ButtonContainerProps` by `BaseButtonProps` in export?

File: modules/react/button/lib/BaseButton.tsx [hunk 2]
```diff
@@ -40,6 +40,11 @@ export interface ButtonContainerProps extends BoxProps, GrowthBehavior {
   fillIcon?: boolean;
 }
 
+/**
+ * Extends all the style properties from Box to our buttons
+ */
+export interface BaseButtonProps extends Omit<ButtonContainerProps, 'ref'> {}
```
Comment (RayRedGoose) [inline modules/react/button/lib/BaseButton.tsx:46]: This has more sense!

File: modules/react/button/stories/button/Button.stories.mdx [hunk 3]
```diff
@@ -117,6 +118,15 @@ Undocumented props are spread to the underlying `<button>` element.
...
 
+### Custom Styles
+
+We understand that there are instances in which consumers might need to adjust styles for specific
+use cases. All of our buttons extend our lower level base button props which extends style
+properties that come from Box. Below are some examples in which you can overwrite styles for all of
+our buttons. Please use this sparaingly and try to adhere to the design system.
+
+<ExampleCodeBlock code={CustomStyles} />
+
```
Comment (RayRedGoose) [inline modules/react/button/stories/button/Button.stories.mdx:129]: That's really cool! It will help a lot. Thanks for adding

**Review by RayRedGoose — COMMENTED** [review]
I have only one question. I guess some our components use `BaseButton`, should they extends `BaseButtonProps`?

File: modules/react/button/index.ts (hunk 1 above)
Comment (mannycarrera4) [reply modules/react/button/index.ts:14]: yep, makes sense to me!

File: modules/react/button/stories/button/examples/CustomStyles.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+const StyledPrimaryButton = styled(PrimaryButton)({
+  height: space.l,
+  padding: space.xxs,
+});
+
+export const CustomStyles = () => (
+  <HStack spacing="s" padding="s">
+    <PrimaryButton
+      padding={space.l}
+      border={'`1px dotted red'}
```
Comment (jaclynjessup) [inline modules/react/button/stories/button/examples/CustomStyles.tsx:35]: i think you need to remove the ` here. When I got to the example I do not see this border

Comment (mannycarrera4) [reply modules/react/button/stories/button/examples/CustomStyles.tsx:35]: oo good catch, thanks!

File: modules/react/button/stories/button/examples/CustomStyles.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      Styled with style properties
+    </PrimaryButton>
+    <PrimaryButton colors={getDropdownColors()} icon={plusIcon} iconPosition="start">
+      Styled with colors prop
+    </PrimaryButton>
+    <StyledPrimaryButton icon={caretDownIcon} iconPosition="end">
+      Overwrite styles with emotion
+    </StyledPrimaryButton>
+  </HStack>
+);
```
[the remark this answers is not in the document]
Comment (mannycarrera4) [reply modules/react/button/stories/button/examples/CustomStyles.tsx]: I can add that!

**Review by jaclynjessup — APPROVED** [review]

**Pull request merged**